client/api/filesystem/v1alpha1/api.proto

```diff
@@ -4,25 +4,156 @@ package v1alpha1;
 
 import "github.com/kubernetes-csi/csi-proxy/client/api/errors.proto";
 
-// FIXME: this is just an almost empty service, not actually implemented; it's just
-// here as an example of what API group and versions will look like. It will actually
-// be implemented in a later patch.
-
 service FileSystem {
-    // PathExists checks if the given path exists on the host.
+    // PathExists checks if the parameter path exists on the host filesystem.
     rpc PathExists(PathExistsRequest) returns (PathExistsResponse) {}
+
+    // Mkdir creates a directory at the requested path on the host filesystem.
+    rpc Mkdir(MkdirRequest) returns (MkdirResponse) {}
+
+    // Rmdir removes the directory at the requested path on the host filesystem.
+    // This may be used for unlinking a symlink created through LinkPath.
+    rpc Rmdir(RmdirRequest) returns (RmdirResponse) {}
+
+    // LinkPath creates a local directory symbolic link between the source path 
+    // and target path on the host filesystem.
+    rpc LinkPath(LinkPathRequest) returns (LinkPathResponse) {}
+}
+
+// Context of the paths used for path prefix validation
+enum PathContext {
+    // Indicates the kubelet-csi-plugins-path parameter of csi-proxy be used as the path context
+    PLUGIN = 0;
+    // Indicates the kubelet-pod-path parameter of csi-proxy be used as the path context
+    CONTAINER = 1;
 }
 
 message PathExistsRequest {
-    // The path to check in the host filesystem.
+    // The path whose existence we want to check in the host filesystem.
     string path = 1;
+
+    // Context of the path parameter.
+    // This is used to determine the root for relative path parameters
+    PathContext context = 2;
 }
 
 message PathExistsResponse {
-    bool success = 1;
+    // The Windows API error code observed (if any)
+    api.WindowsApiError api_error = 2;
```

Comment:
The way I imagined it, `CmdletError` would be used when the RPC call mapped to (and was being run as) a powershell cmdlet, and `WindowsApiError` when the RPC call mapped to a Windows API procedure (e.g. https://docs.microsoft.com/en-us/windows/win32/api/iscsidsc/nf-iscsidsc-addiscsiconnectionw).

Not sure `PathExists` maps to either of these though; presumably that would be just calling `os.Stat` in go? So maybe a simple `string error = 1` would be enough here?

Reply:
And in any case this should be field # 1 , no one out there is running the mock API that's currently in master :)